contrib/browndog/PEcAn#DALEC_convert.R

```diff
@@ -14,75 +14,75 @@ verbose <- TRUE
 }
 
 # parse/check arguments
 if (length(args) < 2) {
-    usage("Not enough arguments")
+  usage("Not enough arguments")
 }
 if (length(args) > 2) {
-    tempDir <- args[3]
+  tempDir <- args[3]
 } else {
-    tempDir <- "temp"    
+  tempDir <- "temp"
 }
 
 inputFile <- args[1]
 outputFile <- args[2]
 
 # create folders
-cffolder <- file.path(tempDir,"cf")
-dir.create(cffolder, showWarnings=FALSE, recursive=TRUE)
+cffolder <- file.path(tempDir, "cf")
+dir.create(cffolder, showWarnings = FALSE, recursive = TRUE)
 
-outfolder <- file.path(tempDir,"linkages")
-dir.create(outfolder, showWarnings=FALSE, recursive=TRUE)
+outfolder <- file.path(tempDir, "linkages")
+dir.create(outfolder, showWarnings = FALSE, recursive = TRUE)
 
 # unzip and parse filenames
 if (grepl("pecan.zip$", args[1])) {
-    system2("/usr/bin/unzip", c("-o", "-d", cffolder, inputFile))
-    site <- NA
-    startYear <- NA
-    endYear <- NA
-    for(file in list.files(path=cffolder, pattern="*.nc")) {
-        pieces <- strsplit(file, ".", fixed=TRUE)[[1]]
-        if (length(pieces) != 3) {
-          usage(paste0("invalid file ", file, " should be <site>.<year>.nc"))
-        }
-        if (is.na(site)) {
-            site <- pieces[1]
-        } else if (site != pieces[1]) {
-            usage(paste0("incosistent sites ", file, " should be ", site, ".<year>.nc"))
-        }
-        if (is.na(startYear) || pieces[2] < startYear) {
-            startYear <- pieces[2]
-        }
-        if (is.na(endYear) || pieces[2] > endYear) {
-            endYear <- pieces[2]
-        }
-        startDate <- as.POSIXlt(paste0(startYear,"-01-01 00:00:00"), tz = "GMT")
-        endDate <- as.POSIXlt(paste0(endYear,"-12-31 23:59:59"), tz = "GMT")
+  system2(Sys.which("unzip"), c("-o", "-d", cffolder, inputFile))
+  site <- NA
+  startYear <- NA
+  endYear <- NA
+  for (file in list.files(path = cffolder, pattern = "*.nc$")) {
+    pieces <- strsplit(file, ".", fixed = TRUE)[[1]]
+    if (length(pieces) != 3) {
+      usage(paste0("invalid file ", file, " should be <site>.<year>.nc"))
     }
-} else if (grepl("pecan.nc$", inputFile)) {
-    pieces <- strsplit(inputFile, ".", fixed=TRUE)[[1]]
-    if (length(piecesx) != 4) {
-      usage("Input file name should be of format <site>.<year>.pecan.nc")
+    if (is.na(site)) {
+      site <- pieces[1]
+    } else if (site != pieces[1]) {
+      usage(paste0("inconsistent sites ", file, " should be ", site, ".<year>.nc"))
+    }
+    if (is.na(startYear) || pieces[2] < startYear) {
+      startYear <- pieces[2]
     }
-    site <- pieces[1]
-    year <- pieces[2]
-    file.copy(inputFile, file.path(cffolder, paste(site, year, "nc", sep=".")))
-    startDate <- as.POSIXlt(paste0(year,"-01-01 00:00:00"), tz = "GMT")
-    endDate <- as.POSIXlt(paste0(year,"-12-31 23:59:59"), tz = "GMT")
+    if (is.na(endYear) || pieces[2] > endYear) {
+      endYear <- pieces[2]
+    }
+    startDate <- as.POSIXlt(paste0(startYear, "-01-01 00:00:00"), tz = "GMT")
```

Comment:
This should be UTC (error in original code). GMT != UTC
Same for the rest of the files.